src/components/services/AccountCallbackManager.js

```diff
@@ -0,0 +1,164 @@
+// @flow
+
+import { type EdgeAccount, type EdgeCurrencyWallet } from 'edge-core-js'
+import { watchSecurityAlerts } from 'edge-login-ui-rn'
+import * as React from 'react'
+
+import { updateExchangeRates } from '../../actions/ExchangeRateActions.js'
+import { checkPasswordRecovery } from '../../actions/RecoveryReminderActions.js'
+import { newTransactionsRequest, refreshTransactionsRequest } from '../../actions/TransactionListActions.js'
+import { updateWalletLoadingProgress, updateWalletsRequest } from '../../actions/WalletActions.js'
+import { useAsyncEffect } from '../../hooks/useAsyncEffect.js'
+import { useWalletsSubscriber } from '../../hooks/useWalletSubscriptions.js'
+import { useEffect, useState } from '../../types/reactHooks.js'
+import { useDispatch } from '../../types/reactRedux.js'
+import { Actions } from '../../types/routerTypes.js'
+import { isReceivedTransaction, snooze } from '../../util/utils.js'
+import { WcSmartContractModal } from '../modals/WcSmartContractModal.js'
+import { Airship } from './AirshipInstance.js'
+
+type Props = {
+  account: EdgeAccount
+}
+
+// Tracks items that need a refresh:
+type DirtyList = {
+  rates: boolean,
+  walletList: boolean,
+  wallets: { [walletId: string]: EdgeCurrencyWallet }
+}
+
+const notDirty: DirtyList = {
+  rates: false,
+  walletList: false,
+  wallets: {}
+}
+
+export function AccountCallbackManager(props: Props) {
+  const { account } = props
+  const dispatch = useDispatch()
+  const [dirty, setDirty] = useState<DirtyList>(notDirty)
+
+  // Helper for marking wallets dirty:
+  function addWallet(wallet: EdgeCurrencyWallet) {
+    setDirty(dirty => ({
+      ...dirty,
+      rates: true, // We might reconsider this behavior
+      wallets: { ...dirty.wallets, [wallet.id]: wallet }
+    }))
+  }
```

Comment:
Nope, `useCallback` would do nothing in this case, because of the way `useWalletsSubscriber` captures its parameter.

Reply:
Like I've mentioned, this comment was in case you change the next one into useCallback, in that case 'addWallet' is a dependency, but since it's not, it's fine.